src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -990,5 +991,28 @@ private void EmbedMetadataReferenceInformation(CommonPEModuleBuilder module)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.CompilationMetadataReferences),
                 value: _debugMetadataOpt.GetOrAddBlob(builder));
         }
+
+        private void TypeDocumentInformation(CommonPEModuleBuilder module)
+        {
+            DocumentHandle rowid;
+            var document = module.GetTypeDocument(Context);
+            if (document.Count > 0)
+            {
+                foreach (var pair in document)
+                {
+                    var builder = new BlobBuilder();
+                    var def = (ITypeDefinition)pair.definition;
```

Comment:
>pair.definition [](http://example.com/codeflow?start=47&length=15)

Here you assume that the definition is ITypeDefinition rather then IDefintion. If that is so why doesn't GetTypeDocument return ITypeDefinitions?